pkg/reconciler/stats_reporter.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"context"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+type Measurement int
+
+const (
+	// ServiceTimeUntilReadyM is the time it takes for a service to become ready since the resource is created
+	ServiceTimeUntilReadyM Measurement = iota
+
+	// ServiceTimeUntilReadyN is the time it takes for a service to become ready since the resource is created
+	ServiceTimeUntilReadyN = "service_time_until_ready"
+)
+
+var (
+	measurements = []*stats.Float64Measure{
+		ServiceTimeUntilReadyM: stats.Float64(
+			ServiceTimeUntilReadyN,
+			"Time it takes for a service to become ready since created",
+			stats.UnitMilliseconds),
+	}
+	reconcilerTagKey tag.Key
+)
+
+
+func init() {
+	var err error
+	// Create the tag keys that will be used to add tags to our measurements.
+	// Tag keys must conform to the restrictions described in
+	// go.opencensus.io/tag/validate.go. Currently those restrictions are:
+	// - length between 1 and 255 inclusive
+	// - characters are printable US-ASCII
+	reconcilerTagKey, err = tag.NewKey("reconciler")
```

Comment:
Would be better to do something like: https://github.com/knative/pkg/blob/1f624b77dd9358af2154fe1cd1e5c03cf8985908/controller/stats_reporter.go#L43